src/app/shared/components/list/list.component.ts

```diff
@@ -22,31 +22,31 @@ export class ListComponent implements OnInit {
   @Input() pagesToLoad: number = 3;
   @Input() loadTime: number = 5000;
 
-  currentPage: number = 1;
-  lastPage: number = this.currentPage;
-  loading: boolean = true;
+  currentPage = 1;
+  lastPage = this.currentPage;
+  loading = false;
   Orientations = Orientation;
 
-  nextPageCallback: () => unknown = this.nextPage.bind(this);
-  previousPageCallback: () => unknown = this.previousPage.bind(this);
+  nextPageCallback = this.nextPage.bind(this);
+  previousPageCallback = this.previousPage.bind(this);
 
   constructor() {}
 
   ngOnInit(): void {
     if (this.items.length === 0) {
       this.loadNext(this.pageSize * (this.paging ? this.pagesToLoad : 1));
     } else {
-      this.lastPage = this.items.length / this.pageSize;
+      this.lastPage = Math.ceil(this.items.length / this.pageSize);
     }
   }
 
-  loadNext(count: number = this.pageSize): void {
+  loadNext(_count: number = this.pageSize): void {
     this.loading = true;
 
-    this.service?.getMany(count, this.loadTime).subscribe((items: (typeof this.service.dataType)[]) => {
+    this.service?.getMany(_count, this.loadTime).subscribe((items: (typeof this['service']['dataType'])[]) => {
       this.items.push(...items);
-      this.lastPage = this.items.length / this.pageSize;
-      this.loading = false;
+      this.lastPage = Math.ceil(this.items.length / this.pageSize);
```

Comment:
good catch

Reply:
yep I figured I use Math.ceil because it didn't make sense to me.